src/btree/bt_vrfy.c

```diff
@@ -132,6 +139,156 @@ __verify_layout(WT_SESSION_IMPL *session, WT_VSTUFF *vs)
     return (0);
 }
 
+/*
+ * __verify_col_var_page_hs --
+ *     Verify a page against the history store.
+ */
+static int
+__verify_col_var_page_hs(WT_SESSION_IMPL *session, WT_REF *ref, WT_VSTUFF *vs)
+{
+    WT_CELL *cell;
+    WT_CELL_UNPACK *unpack, _unpack;
+    WT_COL *cip;
+    WT_DECL_ITEM(key);
+    WT_DECL_RET;
+    WT_PAGE *page;
+    uint64_t recno, rle;
+    uint32_t i;
+    uint8_t *p;
+
+    page = ref->page;
+    recno = ref->ref_recno;
+    unpack = &_unpack;
+
+    /* Ensure enough room for a column-store key without checking. */
+    WT_ERR(__wt_scr_alloc(session, WT_INTPACK64_MAXSIZE, &key));
+
+    WT_COL_FOREACH (page, cip, i) {
+        p = key->mem;
+        WT_ERR(__wt_vpack_uint(&p, 0, recno));
+        key->size = WT_PTRDIFF(p, key->data);
+
+        cell = WT_COL_PTR(page, cip);
+        __wt_cell_unpack(session, page, cell, unpack);
+        rle = __wt_cell_rle(unpack);
+
+#ifdef HAVE_DIAGNOSTIC
+        /* Optionally dump historical time pairs and values in debug mode. */
+        if (vs->dump_history) {
+            WT_ERR(__wt_msg(session, "\tK {%" PRIu64 " %" PRIu64 "}", recno, rle));
+            WT_ERR(__wt_debug_key_value(session, NULL, unpack));
+        }
+#endif
+
+        WT_ERR(__verify_key_hs(session, key, vs));
+        recno += rle;
+    }
+
+err:
+    __wt_scr_free(session, &key);
+
+    return (ret);
+}
+
+/*
+ * __verify_row_leaf_page_hs --
+ *     Verify a page against the history store.
+ */
+static int
+__verify_row_leaf_page_hs(WT_SESSION_IMPL *session, WT_REF *ref, WT_VSTUFF *vs)
+{
+    WT_CELL_UNPACK *unpack, _unpack;
+    WT_DECL_ITEM(key);
+    WT_DECL_RET;
+    WT_PAGE *page;
+    WT_ROW *rip;
+    uint32_t i;
+
+    page = ref->page;
+    unpack = &_unpack;
+
+    WT_RET(__wt_scr_alloc(session, 256, &key));
```

Comment:
Is there an existing #define for the value 256 here?

Reply:
No. I've copied this usage from `bt_debug` in `__debug_page_row_leaf` but other uses of this scratch alloc function just use hardcoded numbers or have a calculated value.

Reply:
Ah okay, that works then.